source/gems/dry-types/1.0/getting-started.html.md

```diff
@@ -0,0 +1,57 @@
+---
+title: Getting Started
+layout: gem-single
+name: dry-types
+---
+
+### Using `Dry::Types` in Your Application
+
+1. Make `Dry::Types` available to the application by creating a namespace that includes `Dry::Types`:
+
+    ```ruby
+    module Types
+     include Dry.Types()
+    end
+    ```
+   
+2. Reload the environment, & type `Types::Coercible::String` in the ruby console to confirm it worked:
+
+    ``` ruby
+    Types::Coercible::String
+    # => #<Dry::Types::Constructor type=#<Dry::Types::Definition primitive=String options={}>>
+    ```
+
+### Creating Your First Type
+
+1. Define a struct's types by passing the name & type to the `attribute` method:
+
+    ```ruby
+    class User < Dry::Struct
+      attribute :name, Types::String
+    end
+    ```
```

Comment:
I'm not sure if we want to show dry-struct here, given that these are docs for dry-types. I know that the projects are often used together but *for now* it would be better to focus on individual gems. I should mention that we'll be addressing cross-gem docs in the next version of the website.

Reply:
What should be here instead? I'm thinking about plain Ruby objects.

It would be great to show a common use-case:

1. I have a _stateful_ object
2. I want its state to be valid at all times
3. So I use `dry-type` in my writers to ensure the proper type

Reply:
@Morozzzko I'm not sure to be honest

Reply:
99% of all use cases for `dry-types` is with `attribute`, a feature that only `dry-struct` and similar modelling libraries have.

With that said, if there's any interest in decoupling `dry-struct` and removing it from the docs, I would be happy to.

Something like...

[ ] Remove references to Dry::Struct in place of basic Constraint examples
[ ] Create a separate documentation in `dry-struct` that could be linked to from the `dry-types` docs that combines all the references I extracted into a definitive page

lmk 👍

Reply:
There's also dry-schema, there's no `attribute` but it's quite similar anyway. I think examples should use dry-struct, otherwise, it's not clear what to do with dry-types. It's quite a fundamental library and is mostly used as part of bigger gems.

Reply:
OK I don't want to block this so let's stick to dry-struct for now. I'll be working on the doc overhaul soon anyway so I'll address this somehow.

Reply:
@solnic feel free to list off any tasks you need for the docs and I'd be happy to implement them. I was thinking the best approach would be to build out the docs using examples of simple Constraint objects, then from there introduce how to use it with `attribute` and modelling/schema libraries. lmk

Reply:
@patrickclery will do, first I need to finish our middleman 5.0 upgrade with a bunch of improvements that will make writing docs simpler and THEN I'll have time to come up with specific tasks.